### PR TITLE
Convergence Tester Update

### PR DESCRIPTION
**Pull request by eslickj**
Updates for convergence tester.  There are a few things things here.  1) update the command line interface and adding tests 2) doing my best address issues in #176 3) moving and registering convergence test classes (the convergence class import path was to long for my typing comfort).  I didn't intentionally change the way anything works, so it should basically function as it always did with a few extra little features and an easier interface. 

I guess we don't really have docs.  The interface is just expanding on @carldlaird's original and adding it to the idaes command.  For now you can use: 

```sh
    idaes convergence-search --help
    idaes convergence-sample --help
    idaes convergence-eval --help
```

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by eslickj** [comment]
Oh, as per issue #176, I added a single sample run option, but I'm not too sure what we want to do with the results of that.  Maybe it's something we can expand on in the future, or if there are ideas, I can fix some stuff up in this PR too.

File: idaes/__init__.py [hunk 1]
```diff
@@ -59,6 +59,22 @@ def _create_testing_dir():
     _create_data_dir()
     config.create_dir(testing_directory)
 
+try:
+    _create_data_dir()
+except FileNotFoundError:
+    pass # the standard place for this doen't exist, shouldn't be a show stopper
```
Comment (andrewlee94) [inline idaes/__init__.py:65]: Typo: doesn;t

Also lines below.

**Review by andrewlee94 — COMMENTED** [review]
I found one minor typo, but I also am not really familiar with the convergence tester code so I am not sure how much else I can contribute.

**Review by Andresj89 — DISMISSED** [review]
I tested this PR with a supercritical power plant flowsheet and everything seems to be working fine.

**Review by Andresj89 — APPROVED** [review]
LGTM

File: idaes/core/util/convergence/convergence.py [hunk 2]
```diff
@@ -1,228 +0,0 @@
...
-         -s PressureChanger-10.json
-
-Note that the convergence evaluation can also be run in parallel if you have
-installed MPI and mpi4py using a command line like the following::
-
-   $ mpirun -np 4 python ../../../core/util/convergence/convergence.py run-eval
-         -s PressureChanger-10.json
-
-"""
-import argparse
```
Comment (carldlaird) [inline idaes/core/util/convergence/convergence.py:60]: John, where did this "documentation" end up?

**Review by carldlaird — COMMENTED** [review]
Please see comment about documentation.

File: idaes/core/util/convergence/convergence.py (hunk 2 above)
Comment (eslickj) [reply idaes/core/util/convergence/convergence.py:60]: If we're talking sphinx docs, I don't know about it.  If we're talking about command line help, its in https://github.com/IDAES/idaes-pse/pull/231/files#diff-ba1405186222d7f01f8af3acb1b8f7f484254714bd53d2b5fb6b1b502548fa93 (idaes/commands/convergence.py), just search for help.

**Review by andrewlee94 — APPROVED** [review]
I will take John E.'s word that this doesn't affect functionality.

**Pull request merged**